### PR TITLE
Big Refinement Update (and small language changes)

### PR DESCRIPTION
**Pull request by odanivan**
- made BaseChooser (ConstraintChooser) into a module which Constraints and Refinements use
- implemented the RefinementItem which also updates the policy correctly (ODRL compliant)
- made little style changes in the natural language interface of the RuleItem
- added language.js for future updates that make natural language interfaces code simpler

### CODE REVIEW

File: src/components/RuleItem.vue [hunk 1]
```diff
@@ -106,12 +107,6 @@ export default {
       required: true,
     },
   },
-  data() {
```
Comment (odanivan) [inline src/components/RuleItem.vue:106]: This was some very old code that we never deleted.

File: src/components/RuleItem.vue [hunk 2]
```diff
@@ -139,25 +134,31 @@ export default {
...
+        return null;
+      }
+      return this.action[0].refinement;
+    },
+    articles() {
+      return articleMapping;
     },
   },
   methods: {
-    furtherPath(nameSegment, indexSegment) {
```
Comment (odanivan) [inline src/components/RuleItem.vue:153]: Some more code that was never used.

File: src/components/RuleItem.vue [hunk 3]
```diff
@@ -193,6 +194,18 @@ export default {
...
       this.constraints.push(null);
     },
+    addRefinement() {
+      if (this.refinements == null) {
+        // action is just a string, but when adding refinements, action becomes an an array
+        let action = this.action;
+        Vue.delete(this.rule, 'action');
+        
+        Vue.set(this.rule, 'action', [{}]);
+        Vue.set(this.action[0], 'rdf:value', action);
```
Comment (odanivan) [inline src/components/RuleItem.vue:204]: There is no possibility to add more than one action, right?
Does that mean that "hardcoding" action[**0**] is ok?

File: src/libs/odrl/constraints.js [hunk 4]
```diff
@@ -1,20 +1,3 @@
-export class Constraint {
```
Comment (odanivan) [inline src/libs/odrl/constraints.js:1]: Finally, after all these years, the moment has come... to delete this awful piece of code. (whoever wrote it lol)

File: src/libs/odrl/refinements.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+export const operandList = [
```
Comment (odanivan) [inline src/libs/odrl/refinements.js:1]: We should think about how to actually load relevant data dynamically for the given action.

File: src/libs/odrl/rules.js [hunk 6]
```diff
@@ -5,7 +5,7 @@ export const RuleTypes = Object.freeze({
     hasParentRule: false,
     subrule: 'duty',
     description: 'ist erlaubt.',
-    descriptionAddition: ['', ''],
```
Comment (odanivan) [inline src/libs/odrl/rules.js:8]: I thought a shorter name would increase readability in RuleItem

Comment (odanivan) [reply src/libs/odrl/rules.js:8]: Also description and addition both contain the postfix '**tion**' which doesn't look nice.

**Review by hkohlsaat — APPROVED** [review]

**Pull request merged**